### PR TITLE
add additional tests

### PR DESCRIPTION
**Pull request by Rup-Narayan-Rajbanshi**
## Changes
- add additional tests

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: project_types/tests/validate_api_calls_test.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        }
+
+        # check for other status code to raise error
+        mock_response.status_code = 500
+        with pytest.raises(ValidateApiCallError):
+            query_osm([12345], {})
+
+    @patch("requests.post")
+    @patch("project_types.validate.api_calls.remove_noise_and_add_user_info")
+    def test_ohsome(self, mock_remove_noise, mock_post):
```
Comment (tnagorra) [inline project_types/tests/validate_api_calls_test.py]: Please check if this requests osm or osmcha

File: project_types/validate/api_calls.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def remove_troublesome_chars(string: str | None):
         return string
     troublesome_chars = {'"': "", "'": "", "\n": " "}
     for k, v in troublesome_chars.items():
-        string = string.replace(k, v)
+        string = string.replace(k, v).strip()
```
Comment (tnagorra) [inline project_types/validate/api_calls.py]: Do we need to change this? Was this implemented from the previous implementation?

File: project_types/tests/validate_api_calls_test.py
Comment (tnagorra) [inline project_types/tests/validate_api_calls_test.py:1]: project_types/validate/tests would be a better place for these tests
@thenav56 what do you think?

File: utils/tests/common_test.py [hunk 3]
```diff
@@ -231,3 +233,22 @@ def test_parse_b64gzjson_to_dict(self):
...
         result = parse_b64gzjson_to_dict(compressed_task)
         assert result == original_json
+
+    def test_generate_tutorial_firebase_id(self):
+        def is_valid_ulid(value: str) -> bool:
+            try:
+                ulid.ULID.from_str(value)
+                return True
+            except ValueError:
+                return False
```
Comment (tnagorra) [inline utils/tests/common_test.py]: We should already have a function for this.

File: utils/tests/common_test.py [hunk 4]
```diff
@@ -231,3 +233,22 @@ def test_parse_b64gzjson_to_dict(self):
...
+
+        tutorial_firebase_id = generate_tutorial_firebase_id()
+        assert tutorial_firebase_id.startswith("tutorial_")
+
+        # check is_valid
+        split_id = tutorial_firebase_id.split("_")
+        assert is_valid_ulid(str(split_id[1])) is True
```
Comment (tnagorra) [inline utils/tests/common_test.py]: The check should instead check that the id starts with "tutorial_" instead.

File: utils/tests/geo_test.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        # test_zoom1_bottom_right
+        assert tile_coords_and_zoom_to_quad_key(1, 1, 1) == "3"
+
+        # test_example_zoom3
+        # Example from explanation
+        assert tile_coords_and_zoom_to_quad_key(3, 5, 3) == "213"
+
+        # test_max_zoom_sample
+        # Arbitrary case at zoom 5
+        assert tile_coords_and_zoom_to_quad_key(10, 12, 5) == "03210"
```
Comment (tnagorra) [inline utils/tests/geo_test.py:64]: Can we also include edge cases? https://wiki.openstreetmap.org/wiki/Zoom_levels
Let's add cases for 22 / 23.

File: utils/tests/geo_test.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        # test with zoom level 1
+        p = lat_long_zoom_to_pixel_coords(0, 0, 1)
+        # At zoom 1, map is 512x512 pixels, middle is 256,256
+        assert p.x == 256
+        assert p.y == 256
+
+        # Test some known lat/lon at zoom 2
+        p = lat_long_zoom_to_pixel_coords(37.7749, -122.4194, 2)  # San Francisco approx
+        assert 0 <= p.x <= 1024
+        assert 0 <= p.y <= 1024
```
Comment (tnagorra) [inline utils/tests/geo_test.py]: Why are we comparing a range?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: utils/tests/common_test.py (hunk 4 above)
Comment (Rup-Narayan-Rajbanshi) [reply utils/tests/common_test.py]: @tnagorra this is already done above.

File: utils/tests/geo_test.py (hunk 5 above)
Comment (Rup-Narayan-Rajbanshi) [reply utils/tests/geo_test.py:64]: @tnagorra Are you talking about zoom level 22, 23??

Comment (tnagorra) [reply utils/tests/geo_test.py:64]: Yes sir!
